BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -2030,5 +1893,57 @@ private float AdjustValue(float min, float max, float value, float step)
 
             return adjusted_value;
         }
+
+        private Device GetDevice()
+        {
+            Device device = _context.Devices[0];
+
+            if (string.IsNullOrEmpty(this.SerialNumber))
+                return device;
+
+            foreach(Device dev in _context.Devices)
+            {
+                if (dev.Info[CameraInfo.SerialNumber] == this.SerialNumber)
+                {
+                    device = dev;
+                    break;
+                }
+            }
+
+            return device;
```

Comment:
I would throw an exception if a device with a specific S/N was requested but not found.

Reply:
fixed